### PR TITLE
test: add GamePlayTests

### PR DESCRIPTION
**Pull request by SilasPeters**
## Description

The test to test all tests!
What it tests exactly, is expressed in the `GamePlayTests.cs` file.

In the process of making logic work, many gameobjects have been given a new layer.

Furthermore, this fixes a bug where an enemy can spawn in starting room.

> The test is not optimised, some logic has been expressed but is not tested or visible with the current seed of the level, and it is not completely visible when the enemies get killed.
> Furthermore, there is a bug in the game where you hit enemies that are not visible or out of range. Please ignore this, for this is out of scope of the game.
> Also, yes the healing works but that is not visible in the current seed. That's simply because the agent is too smart. I will try to make him get damage somehow.

## Testability

Run the tests. The scene is `GamePlayTestScene`.

## Checklist
- [x] Set the proper pull request name
- [x] Merged main into your branch
- [x] Added a scene to the game for your changes

### CODE REVIEW

File: aplib.net-demo/Assets/Materials/Colors/Dark Blue Mat.mat [hunk 1]
```diff
@@ -127,7 +127,7 @@ Material:
     - _ZWrite: 1
     m_Colors:
     - _BaseColor: {r: 0.068629384, g: 0.45377085, b: 0.7132075, a: 1}
-    - _Color: {r: 0.06862936, g: 0.45377082, b: 0.7132074, a: 1}
+    - _Color: {r: 0.06862936, g: 0.45377082, b: 0.7132075, a: 1}
```
Comment (joachimdekker) [inline aplib.net-demo/Assets/Materials/Colors/Dark Blue Mat.mat:130]: image

Comment (QuakeEye) [reply aplib.net-demo/Assets/Materials/Colors/Dark Blue Mat.mat:130]: https://github.com/team-zomsa/aplib.net-demo/assets/74355598/b84ec46c-bb6a-4834-901d-7d6a0d97a66a

**Review by JensSteenmetz — APPROVED** [review]
LGTM!

File: aplib.net-demo/Assets/Testing/AplibTests/GameplayTests.cs [hunk 2]
```diff
@@ -0,0 +1,879 @@
...
+                Rigidbody playerRigidBody = beliefSet.PlayerRigidBody;
+                playerRigidBody.AddForce(playerRigidBody.transform.right * -10, ForceMode.VelocityChange);
+            });
+            PrimitiveTactic stepAsideLeft = new(stepAsideLeftAction, beliefSet =>
+            {
+                Rigidbody playerRigidBody = beliefSet.PlayerRigidBody;
+
+                Vector3 left = playerRigidBody.transform.TransformDirection(Vector3.left);
+                return Physics.Raycast(playerRigidBody.transform.position, left, 0.5f,
+                    ~LayerMask.GetMask("PlayerSelf", "Item", "Ignore Raycast"));
```
Comment (JoenvandeVorle) [inline aplib.net-demo/Assets/Testing/AplibTests/GameplayTests.cs:469]: It is not clear to me why you only need those layers. Maybe add a small comment explaining why

**Review by JoenvandeVorle — CHANGES_REQUESTED** [review]
lgtm overall! Got a few comments

**Review by JoenvandeVorle — APPROVED** [review]
awimo!

**Review by QuakeEye — APPROVED** [review]
image

**Pull request merged**